### PR TITLE
Reload tests for normalizing flow models

### PR DESCRIPTION
**Pull request by ncfrey**
This PR adds model reloading capabilities  to `dc.models.normalizing_flows.NormalizingFlowModel`. Reload tests follow the format for other models. This addresses #2217 in part.

CCing @rbharath @peastman

### CODE REVIEW

File: deepchem/models/tests/test_reload.py [hunk 1]
```diff
@@ -282,6 +282,42 @@ def test_robust_multitask_classification_reload():
...
+  dataset = dc.data.NumpyDataset(X=target_distribution.sample(96))
+  final = nfm.fit(dataset, nb_epoch=1)
+
+  assert nfm.flow.sample().numpy().shape == (2,)
+
+  reloaded_model = NormalizingFlowModel(nf, model_dir=model_dir)
+  reloaded_model.restore()
+
+  # Check that reloaded model can sample from the distribution
+  assert reloaded_model.flow.sample().numpy().shape == (2,)
```
Comment (rbharath) [inline deepchem/models/tests/test_reload.py:318]: For the reload test, we should make sure that `nfm` and `reloaded_model` return the exact same output after the reload is complete. Right now we're only checking shape. Could we do a direct output equality check here as well?

**Review by rbharath — COMMENTED** [review]
Thanks for getting this started! Have a request on a more rigorous test on the reload

File: deepchem/models/tests/test_reload.py (hunk 1 above)
Comment (ncfrey) [reply deepchem/models/tests/test_reload.py:321]: Sounds good! The sampling is random so the best I can do is check shape there, but I added a test to check that density estimation returns the exact same output after the reload.

**Review by rbharath — APPROVED** [review]
Ok this looks good! Feel free to merge when ready :)

**Pull request merged**